### PR TITLE
Studioc improvements

### PR DESCRIPTION
**Pull request by conrad-vanl**
## ✏️ Description

Adds a badge popover for studioc badges. I believe this is better then linking to a new content view: Simpler, self-contained, better UX.

This seeks to solve:
https://3.basecamp.com/3926363/buckets/28712797/todos/5762593053
https://3.basecamp.com/3926363/buckets/28712797/todos/5762645151

The only thing remaining is bringing the badge "Description" text into the popover (commented out in this PR), and confirming that `action.title` is pulling the correct Badge CTA Text from StudioC API.

## 🔬 To Test
1. Switch to Community Christian App
2. Click on badges

## 📸 Screenshots

https://user-images.githubusercontent.com/103927/215946875-3181f4b4-e982-40ba-9721-eb05deacedd2.mp4

## 📝 Checklist

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)
- [x] For PR's with design updates/changes, I have consulted with the product designer and made all necessary adjustments in Figma

### CODE REVIEW

File: packages/apollos-ui-kit/src/ContentTitles/ContentTitles.js [hunk 1]
```diff
@@ -69,11 +69,11 @@ const Container = styled(
   'ui-kit.ContentTitles.Container'
 )(PaddedView);
 
-const renderWithIcon = (actionIcon, label) =>
+const renderWithIcon = (actionIcon, label, ActionIconComponent = ActionIcon) =>
```
Comment (redreceipt) [inline packages/apollos-ui-kit/src/ContentTitles/ContentTitles.js:72]: Why make the component a prop?

File: templates/mobile/src/auth/LandingSwiper/index.js [hunk 2]
```diff
@@ -26,9 +26,7 @@ const LandingSwiper = () => {
   });
   const slideComponents = useMemo(() => {
     // Default to Auth / OpenID landing screen if we have OpenID providers, otherwise direct to IdentityEntry.
-    const defaultAuthScreen = data?.openIdProviders?.length
-      ? 'OpenID'
```
Comment (redreceipt) [inline templates/mobile/src/auth/LandingSwiper/index.js:30]: Can we do this in a separate PR to make sure we take care of all OpenId related stuff at the same time?

File: packages/apollos-ui-kit/src/ContentTitles/ContentTitles.js (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-ui-kit/src/ContentTitles/ContentTitles.js:72]: Nvm I see it

File: templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js [hunk 3]
```diff
@@ -122,35 +144,74 @@ const Badge = ({ isActive, icon, onPress, action }) => {
...
       </AnimatedPressable>
+
+      <BottomSheetModal
+        snapPoints={snapPoints}
+        ref={bottomSheetRef}
+        backgroundStyle={styles.bottomSheet({ theme })}
+        handleIndicatorStyle={styles.handleIndicator({ theme })}
+        stackBehavior="replace"
+      >
+        <SystemThemeProvider theme={theme}>
```
Comment (redreceipt) [inline templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js:173]: We have to drop a provider here? Surely this up the tree somewhere already?

File: templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js [hunk 4]
```diff
@@ -122,35 +144,74 @@ const Badge = ({ isActive, icon, onPress, action }) => {
...
+          <BottomSheetScrollView style={styles.flex}>
+            <ContentTitles
+              title={name}
+              featured
+              ActionIconComponent={() => <BadgeIcon icon={icon} size={64} />}
+              actionIcon={'blank'}
```
Comment (redreceipt) [inline templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js:179]: we should make this optional instead of pass "blank"

File: templates/mobile/src/auth/LandingSwiper/index.js (hunk 2 above)
Comment (conrad-vanl) [reply templates/mobile/src/auth/LandingSwiper/index.js:30]: Oh this wasn't supposed to make it in sorry

Comment (conrad-vanl) [reply templates/mobile/src/auth/LandingSwiper/index.js:30]: i was having trouble locally where OpenID kept popping up so i just turned it off. didn't mean to commit it

File: templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js (hunk 3 above)
Comment (conrad-vanl) [reply templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js:173]: Right, the button wasn't working without it. I'm not sure why. @calebpanza maybe you can help point some light on this?

File: templates/mobile/src/ui-kit/theme/DarkThemeProvider/colorMap.js [hunk 5]
```diff
@@ -10,8 +10,8 @@
 import Color from 'color';
 import { mergeThemeColors } from '../../utils';
 
-function dark(theme) {
-  const { colors } = theme;
+function dark(theme = {}) {
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply templates/mobile/src/ui-kit/theme/DarkThemeProvider/colorMap.js:13]: when i was trouble shooting why the button wasn't working without the provider, didn't mean to commit

File: templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js (hunk 3 above)
Comment (calebpanza) [reply templates/mobile/src/features/BadgeListFeatureConnected/BadgeList/BadgeList.js:173]: It's possible that the context is negated with the BottomSheetModal. I'm not 100% sure, although nesting the Provider like this is 100% an acceptable practice. I'll leave a note for myself to come back to this , though, and see if I can figure out what's up;

**Review by redreceipt — APPROVED** [review]
